pages/index.js

```diff
@@ -43,38 +43,21 @@ export default function Home() {
   const [walletAddress, setWalletAddress] = useState(null)
   const [inputValue, setInputValue] = useState('')
   const [gifList, setGifList] = useState([])
+  const [provider, setProvider] = useState(null)
 
-
-
-  const checkIfWalletIsConnected = async () => {
-    try {
-      const { solana } = window;
-
-      if (solana) {
-        if (solana.isPhantom) {
-          console.log('Phantom wallet found!');
-          const res = await solana.connect({ onlyIfTrusted: true });
-          console.log('Connected with Public Key:', res.publicKey.toString());
-          userWalletAddress = res.publicKey.toString()
-          // setting address to state
-          setWalletAddress(res.publicKey.toString())
-
+  useEffect(() => {
+    if ("solana" in window && window.solana.isPhantom) {
+      // Will either automatically connect to Phantom, or do nothing.
+      try {
+        const pubKey = window.solana.connect({ onlyIfTrusted: true })
+        if (pubKey) {
+          setWalletAddress(pubKey.toString())
         }
-      } else {
-        alert('Solana object not found! Get a Phantom Wallet 👻');
+      } catch (e) {
+        console.error(e)
       }
-    } catch (e) {
-      console.error(e);
     }
-  };
-
-
-  useEffect(() => {
-    window.addEventListener('load', async (event) => {
```

Comment:
no need to add event listener to window here as useEffect watches for changes in deps
since theres an empty array given to deps it'll run given function once on component rendered

that's why I removed